### PR TITLE
Feat/authentication/AuthUserData

### PR DESCRIPTION
**Pull request by coaguila**
# AuthUserData

## Description
This PR introduces new data classes and functions to improve upon the profiles. It closes issue #128 .

## Changes
 - `AuthenticationModel.kt`: Added a new function `currentAuthUser` that returns the `UserInfo` of the currently signed in user
 - `AuthenticationSupabaseModel.kt`: Implemented the new `currentAuthUser` it inherits from `AuthenticationModel.kt`
 - `AuthenticationViewModel.kt`: Added a new state, `authUserData` and a new functiton `loadUserState` that updates said state.
 - `AuthUserData.kt`: New data class created that contains key information for the user profile
 - `AuthenticationSupabaseModelTest.kt`: Updated test by adding tests for `currentAuthUser`.
 - `AuthenticationViewModelTest.kt`: Updated test by adding tests for `loadUserState` and `authUserData`

#### Added
 - `AuthUserData.kt`

#### Modified
 - `AuthenticationModel.kt`
 - `AuthenticationSupabase.kt`
 - `AuthenticationViewModel.kt`
 - `AuthenticationSupabaseModelTest.kt`
 - `AuthenticationViewModelTest.kt`

### CODE REVIEW

**Review by taghizadlaura — APPROVED** [review]
## Summary
LGTM 

## Important

#### Code Quality
- **Readability**: LGTM : The code is generally well-structured and easy to read. Function and variable names are descriptive, which aids in understanding the code's purpose.
- **Consistency**: LGTM : The coding style is consistent throughout, with proper formatting and comments explaining the functionality.

#### Functionality
- **Correctness**: LGTM : The implementation of `currentAuthUser` correctly fetches user data and handles cases where no user is logged in. The method uses callbacks for success and failure scenarios effectively.
- **Edge Cases**: LGTM The edge cases (like handling null user data) are properly managed. However, consider adding more detailed logging or error handling to cover potential unexpected states.

## Testing
LGTM :
- The test cases for `currentAuthUser` and `loadAuthUserData` are well-structured and cover both success and failure scenarios. 

## Documentation
LGTM :
- The code is adequately documented with comments explaining the purpose of functions and data classes.

## Suggestions
- Maybe you can expand the documentation for the `AuthUserData` class.
- Maybe consider implementing additional unit tests for any new edge cases identified during testing.

## Steps before PR approved
LGTM

File: app/src/main/java/com/android/periodpals/model/authentication/AuthenticationModelSupabase.kt [hunk 1]
```diff
@@ -117,4 +118,28 @@ class AuthenticationModelSupabase(
...
+    try {
+      val currentUser: UserInfo? = supabaseAuth.currentUserOrNull()
+      if (currentUser == null) {
+        Log.d(TAG, "currentAuthUser: no user logged in")
+        onFailure(Exception("No User Logged In"))
+      }
+      Log.d(TAG, "currentAuthUser: succesfully retrieved data object")
+      onSuccess(currentUser!!)
+    } catch (e: Exception) {
+      Log.d(TAG, "currentAuthUser: exception thrown: ${e.message} ")
```
Comment (lazarinibruno) [inline app/src/main/java/com/android/periodpals/model/authentication/AuthenticationModelSupabase.kt:141]: Nitpick: Shouldn't it be a `Log.e` since it is an exception?

**Review by lazarinibruno — APPROVED** [review]
## Summary

Thanks for your contribution!

- Code is well documented, making it easier to understand.
  
- Error handling is also well-done.
  

## Improvements

- Adding some short explanatory comments to the tests would improve readablility.
  

## Steps before merge

Besides adding some comments to the tests, all is good.
